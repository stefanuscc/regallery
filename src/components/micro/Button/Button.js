import styles from './Button.module.scss'

export const Button = (props) => {
    const {type = 'button', isIcon=false, isActive=false, className="", onClick, ...attrs} = props
    const classNames = `${styles.button} ${isIcon ? styles.icon : ''} ${isActive ? styles.active : ''} ${className}`

    return (
        <button {...attrs} type={type} className={classNames} onClick={(event) => { onClick(event) }}></button>
    )
}

export default Button