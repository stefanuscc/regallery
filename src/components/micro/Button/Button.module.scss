// Vars
@import "/src/_vars";

.button {
    border: 1px solid;
    overflow: hidden;
    outline: none;

    align-items: center;
    background-color: rgb(255 255 255);
    border-color: rgb(209 213 219);
    border-width: 1px;
    color: rgb(55 65 81);
    display: inline-flex;
    font-size: .875rem;
    font-weight: 500;
    line-height: 1.25rem;
    padding: 0.5rem 1rem;
    position: relative;
    cursor: pointer;

    &.active {
        background-color: $secondary !important;
        border-color: $secondary;
        color: white;
    }

    &.icon {
        padding: 0.5rem 0.7rem;
    }
}