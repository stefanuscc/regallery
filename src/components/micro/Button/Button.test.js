import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';
import "../../../setupTests"

test('Button to match snapshot', () => {
    const component = renderer.create(
        <Button 
            key="1"
            isActive={false}
            className=""
            disabled={false}
            onClick={ () => alert('') }
        >
            A Button
        </Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button label is persistence', () => {
    const button = shallow(
        <Button 
            key="1"
            isActive={false}
            className=""
            disabled={false}
            onClick={ () => alert('') }
        >
            A Button
        </Button>
    );

    expect(button.text()).toEqual('A Button');
});