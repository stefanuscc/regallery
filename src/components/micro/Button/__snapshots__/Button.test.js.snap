// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to match snapshot 1`] = `
<button
  className="button   "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  A Button
</button>
`;
